lib/rule-helpers.js

```diff
@@ -0,0 +1,56 @@
+'use babel';
+
+import { get } from 'request-promise';
+
+const DOCUMENTATION_LIFETIME = 86400 * 1000; // 1 day TODO: Configurable?
+const docsRuleCache = new Map();
+let docsLastRetrieved;
+
+export function takeWhile(source, predicate) {
+  const result = [];
+  const { length } = source;
+  let i = 0;
+
+  while (i < length && predicate(source[i], i)) {
+    result.push(source[i]);
+    i += 1;
+  }
+
+  return result;
+}
+
+// Retrieves style guide documentation with cached responses
+export async function getRuleMarkDown(rule) {
+  if (new Date().getTime() - docsLastRetrieved < DOCUMENTATION_LIFETIME) {
+    // If documentation is stale, clear cache
+    docsRuleCache.clear();
+  }
+  if (docsRuleCache.has(rule)) { return docsRuleCache.get(rule); }
+
+  let rawRulesMarkdown;
+  try {
+    rawRulesMarkdown = await get(`https://raw.githubusercontent.com/troessner/reek/master/docs/${rule}.md`);
+  } catch (x) {
+    return '***\nError retrieving documentation';
+  }
+
+  const byLine = rawRulesMarkdown.split('\n');
+  // eslint-disable-next-line no-confusing-arrow
```

Comment:
With the way of writing this to [follow the rule](https://eslint.org/docs/rules/no-confusing-arrow#rule-details) being so similar, I'd prefer that instead.

Disabling a lint rule should only be done with good reason.